packages/editor/src/components/editor/properties/PropertyItem.tsx

```diff
@@ -27,7 +27,7 @@ export const PropertyItem = ({ value: initValue, onChange, field }: PropertyItem
     const label = field.label!;
     switch (field.type) {
       case 'text':
-        return <InputField label={label} value={toString(value)} onChange={updateValue} />;
+        return <InputWithBrowser label={label} value={toString(value)} onChange={updateValue} />;
```

Comment:
I think instead of replace the text block with this new InputWithBrowser, you should introduce a new block type. 
Have a look at the input block how you can do this.

Then a field can choose if the input field should be a plain input or one with a browser.